Sources/SwiftParserDiagnostics/DiagnosticExtensions.swift

```diff
@@ -68,57 +94,42 @@ extension FixIt.MultiNodeChange {
-    return .makeMissing(tokens.map(Syntax.init), transferTrivia: transferTrivia)
+    return self.makeMissing(tokens.map(Syntax.init), transferTrivia: transferTrivia)
   }
 
   /// If `transferTrivia` is `true`, the leading and trailing trivia of the
   /// removed node will be transferred to the trailing trivia of the previous token.
   static func makeMissing(_ node: (some SyntaxProtocol)?, transferTrivia: Bool = true) -> Self {
-    guard let node = node else {
+    guard let node else {
       return FixIt.MultiNodeChange(primitiveChanges: [])
     }
-    var changes = [FixIt.Change.replace(oldNode: Syntax(node), newNode: MissingMaker().rewrite(node, detach: true))]
-    if transferTrivia {
-      changes += FixIt.MultiNodeChange.transferTriviaAtSides(from: [node]).primitiveChanges
-    }
-    return FixIt.MultiNodeChange(primitiveChanges: changes)
+    return self.makeMissing([node], transferTrivia: transferTrivia)
   }
 
-  /// Transfers the leading and trailing trivia of `nodes` to the previous token
-  /// While doing this, it tries to be smart, merging trivia where it makes sense
-  /// and refusing to add e.g. a space after punctuation, where it usually
-  /// doesn't make sense.
-  private static func transferTriviaAtSides(from nodes: [some SyntaxProtocol]) -> Self {
-    let removedTriviaAtSides = (nodes.first?.leadingTrivia ?? []).merging(nodes.last?.trailingTrivia ?? [])
-    if !removedTriviaAtSides.isEmpty, let previousToken = nodes.first?.previousToken(viewMode: .sourceAccurate) {
-      let mergedTrivia = previousToken.trailingTrivia.merging(removedTriviaAtSides)
-      if previousToken.tokenKind.isPunctuation, mergedTrivia.allSatisfy({ $0.isSpaceOrTab }) {
-        // Punctuation is generally not followed by spaces in Swift.
-        // If this action would only add spaces to the punctuation, drop it.
-        // This generally yields better results.
-        return FixIt.MultiNodeChange()
-      }
-      return FixIt.MultiNodeChange(.replaceTrailingTrivia(token: previousToken, newTrivia: mergedTrivia))
-    } else {
-      return FixIt.MultiNodeChange()
-    }
+  /// Replace present nodes with missing nodes.
+  ///
+  /// If `transferTrivia` is `true`, the leading trivia of the first node and
+  /// the trailing trivia of the last node will be transferred to their adjecent
+  /// tokens.
+  static func makeMissing(_ nodes: [some SyntaxProtocol], transferTrivia: Bool = true) -> Self {
+    self.makeMissing(nodes.map(Syntax.init), transferTrivia: transferTrivia)
   }
 
   /// Replace present nodes with their missing equivalents.
   ///
   /// If `transferTrivia` is `true`, the leading trivia of the first node and
   /// the trailing trivia of the last node will be transferred to their adjecent
   /// tokens.
-  static func makeMissing(_ nodes: [Syntax], transferTrivia: Bool = true) -> Self {
+  private static func makeMissing(_ nodes: [Syntax], transferTrivia: Bool = true) -> Self {
```

Comment:
Do we even need this or could `makeMissing(_ nodes: [some SyntaxProtocol], transferTrivia: Bool)` and `makeMissing(_ nodes: [Syntax], transferTrivia: Bool)` be merged into one function.